packages/cli/commands/import/credentials.ts

```diff
@@ -56,60 +60,104 @@ export class ImportCredentialsCommand extends Command {
 					}
 
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
 					await Db.collections.Credentials!.save(credential);
+
+					const sharedCredential = new SharedCredentials();
+
+					await Db.collections.SharedCredentials!.save(
+						Object.assign(sharedCredential, {
+							user: owner,
+							credentials: credential,
+							role: ownerCredentialRole,
+						}),
+					);
 				}
 			} else {
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-				const fileContents = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
+				const credentials = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
 
-				if (!Array.isArray(fileContents)) {
-					throw new Error(`File does not seem to contain credentials.`);
+				if (!Array.isArray(credentials)) {
+					throw new Error(
+						'File does not seem to contain credentials. Make sure the credentials are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
-					if (typeof fileContents[i].data === 'object') {
+				for (i = 0; i < credentials.length; i++) {
+					if (typeof credentials[i].data === 'object') {
 						// plain data / decrypted input. Should be encrypted first.
-						Credentials.prototype.setData.call(
-							fileContents[i],
-							fileContents[i].data,
-							encryptionKey,
-						);
+						Credentials.prototype.setData.call(credentials[i], credentials[i].data, encryptionKey);
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Credentials!.save(fileContents[i]);
+					await Db.collections.Credentials!.save(credentials[i]);
+
+					const sharedCredential = new SharedCredentials();
+
+					await Db.collections.SharedCredentials!.save(
+						Object.assign(sharedCredential, {
+							user: owner,
+							credentials: credentials[i],
+							role: ownerCredentialRole,
+						}),
+					);
 				}
 			}
 			console.info(`Successfully imported ${i} ${i === 1 ? 'credential.' : 'credentials.'}`);
-			process.exit(0);
+			process.exit();
 		} catch (error) {
-			console.error('An error occurred while exporting credentials. See log messages for details.');
-			logger.error(error.message);
+			console.error('An error occurred while importing credentials. See log messages for details.');
+			if (error instanceof Error) logger.error(error.message);
 			this.exit(1);
 		}
 	}
+
+	private async getInstanceOwner(): Promise<User> {
+		const globalRole = await Db.collections.Role!.findOneOrFail({
+			name: 'owner',
+			scope: 'global',
+		});
+
+		const owner = await Db.collections.User!.findOne({ globalRole });
+
+		if (owner) return owner;
+
+		const user = new User();
+
+		await Db.collections.User!.save(
+			Object.assign(user, {
+				firstName: 'default',
+				lastName: 'default',
+				email: null,
+				password: null,
+				resetPasswordToken: null,
+				...globalRole,
+			}),
```

Comment:
if we do this we should just assign the role and let the database handle any defaults